extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -0,0 +1,834 @@
+
+_Input value sockets_ represent data accessed during the node's execution. For example, it could be arguments of math operations or execution parameters such as iteration count for loop nodes or duration for time-related nodes. These sockets **MUST** either be given a static value in the node object or connected to an output value socket of a different node. The node **MAY** access its input value sockets multiple times during the execution. The runtime **MUST** guarantee that all input value sockets have defined values when the node execution starts.
+
+_Output flow sockets_ represent "`function pointers`" that the node will call to advance the graph execution. For example, bodies and branches of flow control nodes are output flow sockets that drive further execution when certain condition are fulfilled. Output flow sockets **MAY** be unconnected; in such a case graph execution proceeds as if such sockets are no-ops.
+
+_Input flow sockets_ represent "`methods`" that could be called on the node. For example, flow control nodes (such as loops and conditions) usually have an `in` input flow socket that starts node's execution. Additional operations **MAY** also be defined such as `reset` for nodes having an internal state.
+
+Nodes **MAY** be configurable through static properties collectively called "`node's configuration`" that **MAY** affect the node's behavior and the number of its sockets, such as the number of cases for a switch-case control flow node.
+
+Input and output value sockets have associated data types, e.g., floats, integers, booleans, etc.
+
+Node's sockets and configurations are defined by its _type_. Node types follow `domain/operation` naming pattern.
+
+A node is executed when its input flow socket is reached by or when one of its output value sockets is requested by another node. Usually, the node executes its dependencies (if any), its own logic, and any number (including zero) of outgoing flow sockets.
+
+[[types]]
+== Value Types
+
+TBD
+
+[[events]]
+== Custom Events
+
+TBD
+
+[[variables]]
+== Variables
+
+TBD
+
+== Node Types
+
+=== Math Nodes
+
+In this section, `floatN` is a placeholder for any of `float`, `float2`, `float3`, `float4`, or `float4x4` types. All value sockets of `floatN` types have the same type within a node.
+
+==== Constants
+
+===== E
+
+[cols="1h,1,2"]
+|===
+| Type | `math/e` | Euler's number
+| Output value sockets | `float value` | 2.718281828459045
+|===
+
+===== Pi
+
+[cols="1h,1,2"]
+|===
+| Type | `math/pi` | Ratio of a circle's circumference to its diameter
+| Output value sockets | `float value` | 3.141592653589793
+|===
```

Comment:
If we have Pi, we should also have Tau. https://tauday.com/tau-manifesto The value Tau is the circle constant (equal to 2*Pi, or rather, Pi is half of Tau). Using Tau usually results in more readable code. Tau is supported in many programming languages such as [C#](https://learn.microsoft.com/en-us/dotnet/api/system.math.tau?view=net-7.0), [Python](https://docs.python.org/3/library/math.html#math.tau), [GDScript](https://docs.godotengine.org/en/stable/classes/class_@gdscript.html#constants), [Unreal Blueprints](https://docs.unrealengine.com/5.3/en-US/BlueprintAPI/Math/Trig/GetTAU/), and more, so it's useful for interoperability with other languages, especially Unreal Blueprints which is conceptually similar.

```
===== Tau

[cols="1h,1,2"]
|===
| Type | `math/tau` | The circle constant, the circumference of the unit circle in radians.
| Output value sockets | `float value` | 6.2831853071795862
|===
```

Reply:
In case more justification is needed, the KHR_interactivity document currently contains Degrees-To-Radians (`math/rad`) and Radians-To-Degrees (`math/deg`). These allow converting between degrees and radians. Tau can be used for a similarly important purpose, dividing or multiplying by this number allows converting between turns and radians, where 1 turn is τ radians. Tau can be used for more than just this, but the point is, if `math/rad` and `math/deg` are justified as a part of KHR_interactivity, then `math/tau` is also justified.